Persist dataset access properties when using H5Fstart_swmr_write

fortnern opened the pull request:
This fix addresses an issue where using H5Fstart_swmr_write would wipe out the access properties for all open datasets. This PR modifies H5Fstart_swmr_write to cache the DAPL ids for all open datasets then use these property lists when closing/reopening the datasets for SWMR reasons.

fortnern committed:
Fix HDFFV-11308 - dataset access properties disappear when using
H5Fstart_swmr_write.

github-actions[bot] committed:
Committing clang-format changes

qkoziol reviewed (approved):
Library change looks fine, but would be nice if it had a regression test also.

fortnern committed:
Add test for H5Fstart_swmr_write() persisting DAPL settings.  Fix bugs
in the library exposed by this test.

github-actions[bot] committed:
Committing clang-format changes

fortnern commented:
Added a regression test, and fixed some bugs exposed by the test. Quincey, do you know why H5Dget_access_plist() (and similar functions) works the way it does, where it copies the DAPL used to open the dataset, then goes over all the properties one by one and retrieves them from the dataset struct? Seems redundant to me.

fortnern commented:
Github tests are hung. I ran the code tests manually and it seems to pass, as did codespell.

derobins commented:
Oops. Clicked wrong button.
